### ISSUE TITLE
`pub build` complains about generic type in const

### ISSUE THREAD

**Issue by zoechi** [dart-lang/observable#31]
https://github.com/dart-lang/observable/blob/master/lib/src/collections/observable_set.dart#L159
```dart
  @override
  Iterator<E> get iterator => const <E>[].iterator;
                                     ^
```
See also dart-lang/sdk#28551

**Comment by zoechi** [comment]
Friendly ping. The changes are not yet published to pub.

**Comment by zoechi** [comment]
```bash
dart2js web/main.dart
```
in https://github.com/bwu-dart-playground/dartsdk_issue_28780
reproduces the error for me

```
 (master) $ dart2js web/main.dart 
lib/model/project.dart:6:9:
Hint: Library 'package:fhir_designer/designer.dart' doesn't export a 'DataConnection' declaration.
Try removing 'DataConnection' from the 'show' clause.
        DataConnection,
        ^^^^^^^^^^^^^^
lib/model/project.dart:7:9:
Hint: Library 'package:fhir_designer/designer.dart' doesn't export a 'DataConnectionReference' declaration.
Try removing 'DataConnectionReference' from the 'show' clause.
        DataConnectionReference,
        ^^^^^^^^^^^^^^^^^^^^^^^
lib/model/project.dart:8:9:
Hint: Library 'package:fhir_designer/designer.dart' doesn't export a 'Field' declaration.
Try removing 'Field' from the 'show' clause.
        Field,
        ^^^^^
lib/model/project.dart:9:9:
Hint: Library 'package:fhir_designer/designer.dart' doesn't export a 'FieldGroup' declaration.
Try removing 'FieldGroup' from the 'show' clause.
        FieldGroup,
        ^^^^^^^^^^
lib/model/project.dart:10:9:
Hint: Library 'package:fhir_designer/designer.dart' doesn't export a 'FieldGroupElement' declaration.
Try removing 'FieldGroupElement' from the 'show' clause.
        FieldGroupElement,
        ^^^^^^^^^^^^^^^^^
lib/model/project.dart:11:9:
Hint: Library 'package:fhir_designer/designer.dart' doesn't export a 'Form' declaration.
Try removing 'Form' from the 'show' clause.
        Form,
        ^^^^
lib/model/project.dart:12:9:
Hint: Library 'package:fhir_designer/designer.dart' doesn't export a 'Menu' declaration.
Try removing 'Menu' from the 'show' clause.
        Menu,
        ^^^^
lib/model/project.dart:13:9:
Hint: Library 'package:fhir_designer/designer.dart' doesn't export a 'MenuItem' declaration.
Try removing 'MenuItem' from the 'show' clause.
        MenuItem,
        ^^^^^^^^
lib/model/project.dart:16:9:
Hint: Library 'package:fhir_designer/designer.dart' doesn't export a 'SearchForm' declaration.
Try removing 'SearchForm' from the 'show' clause.
        SearchForm,
        ^^^^^^^^^^
lib/model/property_metadata.dart:15:37:
Error: Not a compile-time constant.
            displayWidth != null && displayWidth >= 0 ? displayWidth : 35, // TODO(zoechi) https://github.com/dart-lang/sdk/issues/28780
                                    ^^^^^^^^^^^^^^^^^
lib/model/property_metadata.dart:15:37:
Error: Not a compile-time constant.
            displayWidth != null && displayWidth >= 0 ? displayWidth : 35, // TODO(zoechi) https://github.com/dart-lang/sdk/issues/28780
                                    ^^^^^^^^^^^^^^^^^
lib/model/property_metadata.dart:15:37:
Error: Not a compile-time constant.
            displayWidth != null && displayWidth >= 0 ? displayWidth : 35, // TODO(zoechi) https://github.com/dart-lang/sdk/issues/28780
                                    ^^^^^^^^^^^^^^^^^
lib/model/property_metadata.dart:15:37:
Error: Not a compile-time constant.
            displayWidth != null && displayWidth >= 0 ? displayWidth : 35, // TODO(zoechi) https://github.com/dart-lang/sdk/issues/28780
                                    ^^^^^^^^^^^^^^^^^
lib/model/property_metadata.dart:15:37:
Error: Not a compile-time constant.
            displayWidth != null && displayWidth >= 0 ? displayWidth : 35, // TODO(zoechi) https://github.com/dart-lang/sdk/issues/28780
                                    ^^^^^^^^^^^^^^^^^
lib/model/property_metadata.dart:15:37:
Error: Not a compile-time constant.
            displayWidth != null && displayWidth >= 0 ? displayWidth : 35, // TODO(zoechi) https://github.com/dart-lang/sdk/issues/28780
                                    ^^^^^^^^^^^^^^^^^
Error: Compilation failed.
```
```
Dart VM version: 1.23.0-edge.bb72892e335f2ab69aeb0ef3dec5d7f150c80953 (Fri Feb 17 16:16:58 2017) on "linux_x64"
```

**Comment by zoechi** [comment]
@matanlurey @kevmoo Not sure if this is an oversight or if there is a reason this was not published to pub.dartlang.org yet.

**Comment by kevmoo** [comment]
@matanlurey Can we run w/ this?

**Comment by matanlurey** [comment]
Published `0.20.4`.

**Comment by zoechi** [comment]
Great :)
